### ISSUE TITLE
AddOpenTelemetry not working with singleton TracerFactory

### ISSUE THREAD

**Issue by longility** [open-telemetry/opentelemetry-dotnet#567]
I was trying to debug on why traces were not sent to jaeger, and I was able to narrow down the issue.

`services.AddOpenTelemetry(..)` registers `TracerFactory` as singleton, which doesn't work. I tested this by not using `AddOpenTelemetry` extension and registering the `TracerFactory` directly as singleton. When I changed this to scoped lifetime, it works.

**Works**
```csharp
services.AddSingleton(_ =>
                TracerFactory.Create(
                    builder => builder.UseJaeger(o =>
                    {
                        o.ServiceName = "TracingExampleApp";
                        o.AgentHost = "tracing";
                        o.AgentPort = 6831;
                    })));
```
**Does Not Work**
```csharp
services.AddScoped(_ =>
                TracerFactory.Create(
                    builder => builder.UseJaeger(o =>
                    {
                        o.ServiceName = "TracingExampleApp";
                        o.AgentHost = "tracing";
                        o.AgentPort = 6831;
                    })));
```
Code sample: https://github.com/longility/TracingExampleApp/tree/di-tracer-factory-issue

Either the lifetime needs to be changed to scoped or something else needs to change to where traces are sent. That is my guess unless I'm misunderstanding how this should work.

**Comment by MikeGoldsmith** [comment]
That's interesting. TraceFactory should function as a singleton so I'm unsure why it doesn't work.

Can you elaborate on what failure you were seeing when `TracerFactory` was registered a singleton?

**Comment by longility** [comment]
I think the root cause isn't `TracerFactory` since I had time this morning to dig deeper. I think I narrowed it down to `JaegerUdpBatcher.cs` related to flushing I think.

I forced it to flush and can see in jaeger so I think it has to do with the process of flushing.

I did this to force flush to see in jaeger:
```csharp
internal class DebuggingSpanProcessor : SpanProcessor
    {
        private readonly SpanExporter spanExporter;

        public DebuggingSpanProcessor(SpanExporter spanExporter)
        {
            this.spanExporter = spanExporter;
        }

        public override void OnEnd(SpanData span)
        {
            spanExporter.ExportAsync(new[] { span }, default);

            //DEBUG TO FORCE FLUSH
            spanExporter.ShutdownAsync(default).GetAwaiter().GetResult();
        }

        public override void OnStart(SpanData span)
        {
            spanExporter.ExportAsync(new[] { span }, default);
        }

        public override Task ShutdownAsync(CancellationToken cancellationToken)
        {
            return spanExporter.ShutdownAsync(cancellationToken);
        }
    }
```

The `SpanExporter` is `JaegerTraceExporter`.

I don't have time right now, but the next step to debug would be for me to bring `JaegerUdpBatcher` code so I can step through and see how the timer is functioning. However, y'all may be able to identify the issue quicker than me.

**Comment by longility** [comment]
I think the issue is I may not have the latest version that includes this https://github.com/open-telemetry/opentelemetry-dotnet/pull/479
 I have version `0.2.0-alpha.179`, which is the latest at nuget.org. I tried using this feed: https://www.myget.org/F/opentelemetry/api/v3/index.json, however, it isn't allowing me to pull the nuget package. It is stalling and continuously trying to update nuget package but never finishes.

**Comment by MikeGoldsmith** [comment]
Thanks again for your investigation and update.

I'm not sure why you're having trouble with getting a newer nuget from MyGet - are you trying to get it directly or have you added it as a nuget source?

**Comment by longility** [comment]
Of course, I think it might have to do with VS for mac, not sure. It is really weird. I was able to get it working by manually changing the version in the csproj file.

I'm now on `0.2.0-alpha.227`, but I'm getting a different error:
![image](https://user-images.githubusercontent.com/720635/78394522-6b6a2100-75b1-11ea-9496-d837342a3c95.png)

**Comment by MikeGoldsmith** [comment]
@longility I've raised issue #589 to track the service name being set correctly in Jaeger as described above.
